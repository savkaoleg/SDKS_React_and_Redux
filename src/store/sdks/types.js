export const SET_SDKS = 'SET_SDKS'
