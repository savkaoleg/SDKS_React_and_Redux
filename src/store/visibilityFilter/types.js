export const HANDLE_CHANGE = 'HANDLE_CHANGE'
export const HANDLE_SEARCH_CHANGE = 'HANDLE_SEARCH_CHANGE'
