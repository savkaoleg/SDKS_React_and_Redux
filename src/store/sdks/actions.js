import { SET_SDKS } from './types'

export function setSdks(payload) {
  return {
    type: SET_SDKS,
    payload
  }
}
