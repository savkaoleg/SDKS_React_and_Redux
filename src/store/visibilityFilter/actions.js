import { HANDLE_CHANGE, HANDLE_SEARCH_CHANGE } from './types'

export function handleChange(newType, newAditional) {
  return {
    type: HANDLE_CHANGE,
    newType,
    newAditional
  }
}

export function handleSearcChange(newType, searchQuery) {
  return {
    type: HANDLE_SEARCH_CHANGE,
    newType,
    searchQuery
  }
}
