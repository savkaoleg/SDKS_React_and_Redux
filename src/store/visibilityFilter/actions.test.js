import expect from 'expect'
import { handleChange, handleSearcChange } from './actions'
import { HANDLE_CHANGE, HANDLE_SEARCH_CHANGE } from './types'

describe('>>> visibilityFilter > actions', ()=>{

  describe('>>> visibilityFilter > actions > handleChange', ()=>{

    const newType = 'Tag'
    const newAditional = 'TagName'

    it('handleChange snapshot', () => {

      const handleChangeAction = handleChange(newType, newAditional)
      expect(handleChangeAction).toMatchSnapshot()
    })

    it('handleChange', () => {


      const handleChangeAction = handleChange(newType, newAditional)
      expect(handleChangeAction).toEqual({type: HANDLE_CHANGE, newType, newAditional})
    })

  })
  describe('>>> visibilityFilter > actions > handleSearcChange', ()=>{
    const newType = 'Search'
    const searchQuery = 'searchQuery'

    it('handleSearcChange snapshot', () => {

      const handleSearcChangeAction = handleSearcChange(newType, searchQuery)
      expect(handleSearcChangeAction).toMatchSnapshot()
    })

    it('handleSearcChange', () => {


      const handleSearcChangeAction = handleSearcChange(newType, searchQuery)
      expect(handleSearcChangeAction).toEqual({type: HANDLE_SEARCH_CHANGE, newType, searchQuery})
    })
  })
})
