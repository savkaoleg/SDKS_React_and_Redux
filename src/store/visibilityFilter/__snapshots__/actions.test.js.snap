// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>> visibilityFilter > actions >>> visibilityFilter > actions > handleChange handleChange snapshot 1`] = `
Object {
  "newAditional": "TagName",
  "newType": "Tag",
  "type": "HANDLE_CHANGE",
}
`;

exports[`>>> visibilityFilter > actions >>> visibilityFilter > actions > handleSearcChange handleSearcChange snapshot 1`] = `
Object {
  "newType": "Search",
  "searchQuery": "searchQuery",
  "type": "HANDLE_SEARCH_CHANGE",
}
`;
